File: simple_server.py
#!/usr/bin/env python3
"""
Simple Test Server
==================

Basic Flask server to test if the web interface works
"""

from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/')
def index():
    return jsonify({"message": "AI Dungeon Master Server is running!"})

@app.route('/api/health')
def health():
    return jsonify({
        "status": "healthy",
        "message": "Server is running",
        "version": "1.0.0"
    })

@app.route('/api/test')
def test():
    return jsonify({
        "success": True,
        "message": "Test endpoint working"
    })

if __name__ == '__main__':
    print("🚀 Starting simple test server...")
    print("Access at: http://localhost:8080")
    app.run(host='0.0.0.0', port=8080, debug=True) 

File: simple_server.js
const express = require("express");
const cors = require("cors");
const fs = require("fs");
const path = require("path");

const app = express();

// CORS middleware
app.use(cors());
app.use(express.json());

// Serve static files
app.use(express.static(__dirname));

// Simple scenarios data
const getScenarios = () => {
  const scenariosFile = path.join(__dirname, 'data', 'scenarios.json');
  if (fs.existsSync(scenariosFile)) {
    return JSON.parse(fs.readFileSync(scenariosFile, 'utf8'));
  }
  return [];
};

// Scenarios API endpoint
app.get("/api/scenarios", async (req, res) => {
  try {
    const scenarios = getScenarios();
    res.json({
      success: true,
      scenarios: scenarios,
      total: scenarios.length,
    });
  } catch (error) {
    console.error("Get scenarios error:", error);
    res.status(500).json({ error: error.message });
  }
});

// Health check endpoint
app.get("/health", (req, res) => {
  res.json({
    status: "healthy",
    timestamp: new Date().toISOString(),
    server: "Simple AI Dungeon Master API",
    version: "1.0.0",
  });
});

const PORT = process.env.PORT || 5005;

app.listen(PORT, () => {
  console.log(`🚀 Simple AI Dungeon Master Server running on port ${PORT}`);
  console.log(`🔧 API endpoints available at http://localhost:${PORT}/api`);
  console.log(`📊 Scenarios endpoint: http://localhost:${PORT}/api/scenarios`);
  console.log(`🧪 Test page: http://localhost:${PORT}/test_scenario_selection.html`);
});
